protos/table.proto

```diff
@@ -357,21 +357,21 @@ message DataFile {
 //
 // The path of the deletion file is constructed as:
 //   {root}/_deletions/{fragment_id}-{read_version}-{id}.{extension}
-// where {extension} is `.arrow` or `.bin` depending on the type of deletion.
+// where {extension} depends on DeletionFileType.
 message DeletionFile {
-  // Type of deletion file, which varies depending on what is the most efficient
-  // way to store the deleted row offsets. If none, then will be unspecified. If there are
-  // sparsely deleted rows, then ARROW_ARRAY is the most efficient. If there are
-  // densely deleted rows, then BIT_MAP is the most efficient.
+  // Type of deletion file, intended as a way to increase efficiency of the storage of deleted row
+  // offsets. If there are sparsely deleted rows, then ARROW_ARRAY is the most efficient. If there
+  // are densely deleted rows, then BITMAP is the most efficient.
   enum DeletionFileType {
-    // Deletion file is a single Int32Array of deleted row offsets. This is stored as
-    // an Arrow IPC file with one batch and one column. Has a .arrow extension.
+    // A single Int32Array of deleted row offsets, stored as an Arrow IPC file with one batch and
+    // one column. Has a .arrow extension.
     ARROW_ARRAY = 0;
-    // Deletion file is a Roaring Bitmap of deleted row offsets. Has a .bin extension.
+    // A Roaring Bitmap of deleted row offsets. Has a .bin extension.
     BITMAP = 1;
   }
 
-  // Type of deletion file. If it is unspecified, then the remaining fields will be missing.
+  // Type of deletion file. Because in Proto3 enums without an optional cardinality marker does not
+  // have "explicit presence", an absent value is actually a 0, which is ARROW_ARRAY.
```

Comment:
I think we use the proto3 a bit differently, as you see we still put explicit optional annotation. Fields without optional is basically required. This translates to a field having Option or not in the rust generated code.